server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -4344,11 +4356,6 @@ public UserVmVO doInTransaction(TransactionStatus status) throws InsufficientCap
                     } else {
                         vm.setDetail(key, customParameters.get(key));
                     }
-
-                    if (key.equalsIgnoreCase(ApiConstants.BootType.UEFI.toString())) {
```

Comment:
I think we need to add this too, if template details do not have any data, we need to set the deploy VM param into VM details.

Reply:
This is done in line 4345 without any explicit need for an if condition